test/api.jl

```diff
@@ -210,4 +210,15 @@ end
     end end
 end
 
+@testset "Pkg.API.check_package_name: Error message if package name ends in .jl" begin
+    @test_throws Pkg.Types.PkgError Pkg.API.check_package_name("Example.jl")
+    ex = try
+        Pkg.API.check_package_name("Example.jl")
+    catch e
+        e
+    end
+    @test ex isa Pkg.Types.PkgError
+    @test ex.msg == "`Example.jl` is not a valid package name. Perhaps you meant `Example`"
```

Comment:
I believe this should work to test the message too, right?
```suggestion
    @test_throws Pkg.Types.PkgError("`Example.jl` is not a valid package name. Perhaps you meant `Example`") Pkg.API.check_package_name("Example.jl")
```